server/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -270,4 +272,10 @@ protected Set<String> responseParams(RestApiVersion restApiVersion) {
         return responseParams();
     }
 
+    protected static void setErrorTraceTransportHeader(RestRequest r, ThreadContext c) {
+        // set whether data nodes should send back stack trace based on the `error_trace` query parameter
+        if (c != null) {
+            c.putHeader("error_trace", r.param("error_trace", ERROR_TRACE_DEFAULT));
```

Comment:
Why are we setting the default? Aren't we now blowing up the size of every transport message for the default use case?

Reply:
removed from header if error_trace is false and added transport version to set the default when reading it in https://github.com/elastic/elasticsearch/pull/118266/commits/cf16266dea83b2d396385d041b30bd09ea5e66f0